src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/GerritTrigger.java

```diff
@@ -901,6 +908,92 @@ public boolean equals(Object obj) {
         return false;
     }
 
+    /**
+     * Get server by change.
+     *
+     * @param change the change
+     * @return Gerrit server.
+     */
+    private GerritServer getServerByChange(Change change) {
+        for (GerritServer server : PluginImpl.getServers_()) {
+            if (change.getUrl().contains(server.getHostName())) {
+                logger.trace("chnage: {}; server: {}", change, server.getHostName());
+                return server;
+            }
+        }
+        logger.error("can not find server for change: {}", change);
+        return null;
+    }
+
+    /**
+     * Should we trigger on this change?
+     *
+     * @param change the incoming change
+     * @param project the configured gerrit project
+     * @return true if we should.
+     */
+    private boolean isChangeInteresting(Change change, GerritProject project) {
+        boolean shouldTrigger = false;
+        boolean containsFilePathsOrForbiddenFilePaths = ((project.getFilePaths() != null
+                && project.getFilePaths().size() > 0)
+                || (project.getForbiddenFilePaths() != null && project.getForbiddenFilePaths().size() > 0));
+
+        if (isFileTriggerEnabled() && containsFilePathsOrForbiddenFilePaths) {
+            if (project.isInteresting(change.getProject(), change.getBranch(), change.getTopic(),
+                    change.getFiles(new GerritQueryHandler(getServerByChange(change).getConfig())))) {
```

Comment:
Instead of doing the somewhat brittle `getServerBychange` you could return a `Map<Change, GerritQueryHandler>` from `getChanges` since in there you already know which querry handler was used for what change.

Reply:
This PR depends on another PR in gerrit-events (https://github.com/sonyxperiadev/gerrit-events/pull/104), please review, thanks